src/System.Net.Http/src/System/Net/Http/ReadOnlyMemoryContent.cs

```diff
@@ -0,0 +1,38 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Threading.Tasks;
+
+namespace System.Net.Http
+{
+    public sealed class ReadOnlyMemoryContent : HttpContent
+    {
+        private readonly ReadOnlyMemory<byte> _content;
+
+        public ReadOnlyMemoryContent(ReadOnlyMemory<byte> content)
+        {
+            _content = content;
+            if (content.DangerousTryGetArray(out ArraySegment<byte> array))
+            {
+                SetBuffer(array.Array, array.Offset, array.Count);
```

Comment:
I'm sure this is important for some reason but I can't remember what it is.  Can we add a comment?

Reply:
It just allows for some optimized paths (which we can probably refactor subsequently) that avoid going through the rest of the members of the HttpContent to buffer the data.  I'll add a comment.